common/Rhyolite/App.hs

```diff
@@ -23,17 +29,88 @@ import GHC.Generics (Generic)
 import Reflex.Query.Class
 import Data.These
 import Data.Align
+import Data.Functor.Identity
+import Data.Functor.Const
+import Data.Semigroup (First(..))
+import Data.Functor.Compose
+import Data.Constraint.Extras
+import Data.Dependent.Sum
+import Data.GADT.Compare
+import qualified Data.Dependent.Map.Monoidal as DMap
+import Data.Map (Map)
+import qualified Data.Map.Monoidal as MMap
+import qualified Data.Dependent.Map as DMap'
+import Data.Proxy
+import Data.These.Combinators
+import Data.Vessel
+import Data.Vessel.Internal (maybeToThese, VSum(..))
 
 -- | Set-subtraction operation for queries.
 class PositivePart q where
   positivePart :: q -> Maybe q -- ^ Filter a query to only those parts which are selected a positive amount.
 
+instance PositivePart SelectedCount where
+  positivePart x
+    | x > 0 = Just x
+    | otherwise = Nothing
+
+instance PositivePart () where
+  positivePart _ = Nothing
```

Comment:
Is this right?

Reply:
Don't we map `SelectedCount > 0` to `()`

Reply:
Since this is supposed to be the projection to the interior of the positive cone of an ordered group, as such this is the only instance that makes sense.  It might still be better to just not have it though.  @Dan